src/runtime/ndarray.cc

```diff
@@ -76,15 +78,28 @@ struct NDArray::Internal {
     }
     delete ptr;
   }
+
+  static void BufferDeleter(NDArray::Container* ptr) {
```

Comment:
Ideally, we want to instead move these logics into allocator->EmptyNDArray(args), so there is no reverse dependency on the memory manager.  Memory manager just create an instance of NDArray.

You might have to duplicate the Internal::Create logic for a bit, but not by too much. This way memory_manager is just something a developer can choose to add to the runtime or remove